v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -724,9 +711,13 @@ func (r *AzureDeploymentReconciler) getStatus(ctx context.Context, id string) (g
 
 	// Fill the kube status with the results from the arm status
 	// TODO: The owner parameter here should be optional
-	err = status.PopulateFromARM(knownOwner, reflecthelpers.ValueOfPtr(armStatus)) // TODO: PopulateFromArm expects a value... ick
-	if err != nil {
-		return nil, zeroDuration, errors.Wrapf(err, "converting ARM status to Kubernetes status")
+	if s, ok := status.(genruntime.FromARMConverter); ok {
```

Comment:
Why not have `genruntime.NewEmptyVersionedStatus` return a `genruntime.FromARMConverter`?

Reply:
We'd still need a cast, just elsewhere - the `SetStatus()` method on the `KubernetesResource` interface takes a `ConvertibleStatus`.

Reply:
And we can't combine the interfaces because different concerns, and different implementation scopes.